consumers/notifiers/README.md

```diff
@@ -0,0 +1,23 @@
+# Notifiers service
```

Comment:
Wait... I was not expect a full-blown service called `Notifiers`, as we do not have a special service called `Writers`. We have interface for Writers, but then each Writer is a separate service.

While `writers` is a folder that holed different writers, and not a special Writers service, here it looks like "the main boss in town" is a service (Notify service), and notifiers are only "executors" of this service, configured on deployment. It is not same for Writers - you do not configure `Writers` service to use a special writer - writer itself is a service.

So I think that we should re-discuss this naming. I find it also somehow misleading because it suggest that `writers` and `notifiers` have the same structure, when it looks that they do not.

Reply:
@drasko For me, It's actually the same as a writer, a consumer interface impl. Since it is more complex than it looks like a `full-blown` service.
The difference is notifier service, repo, and obviously API (it can't use shared `consumers/api` as writers, cause it's not the same, it's a different API and domain level. Actually it's just the same pattern, nothing else). 
`Wait... I was not expecting a full-blown service called Notifiers, as we do not have a special service called Writers. We have an interface for Writers, but then each Writer is a separate service.` True, as you can see there is a separate service called `smtp-notifier`, it's just one of many notifiers, exactly the same as writers.
@dusanb94 Maybe it will be easy to understand and follow the code if you move `consumers/api` into the `consumers/writers`? That way notifiers and writers would be aligned and easier to understand. I am right? :)

Reply:
They do not have the same structure. They only share the consumer part, that is: both writers and notifier service subscribe to the NATS channel, optionally transform, and consume received messages. While writers can store messages directly, the Notifier service needs to do more: it needs to check the topic, takes all the users who subscribed to that topic, and send them notifications. This flow is common for all the notifiers (just like "take a message and store it" is common for all the writers), but the Notifier itself requires some sort of wrapper to provide HTTP API and subscription management. For that reason, the code for sending notifications is extracted as a separate dependency (Notifier) for `Notifications service`: you implement repository and service only once and reuse it with different Notifiers. Fundamentally, the difference between Notifier and Writer is precisely what you wrote: `you do not configure Writers service to use a special writer - writer itself is a service`, while Notifier can't be a service per se, because it needs supporting layer for subscription handling and API part to make it work. Since that supporting layer is always the same, it can be extracted to the Notifier service.

Reply:
Nicly explained :clap: IMHO it's clean and I don't see any anti-patterns or pitfalls.

Reply:
Thanks! The only problem I see is starting up the service. If we make the Notifier type configurable from the env, it will get complicated because one Notifier service instance needs to handle env vars for all the Notifiers, and different notifiers are configured completely differently. For example, the Notifiers service needs to handle SMTP, SMPP, and WS configurations to support 3 Notifier types. There are 3 approaches:

1. Create multiple main methods for different Notifier types
2. Make Notifier type configurable in `main.go`: all the possible configs need to be handled, but only one Notifier will be created and passed to the service
3. Support multiple Notifiers; make dependency _a list of Notifiers you can pass to the Notification service_; this also implies that in `main` we handle all the configurations

Reply:
Option 1. Is how we did for writers, right? And IMHO it should be merged to a single main for writers, cause the storage layer (e.g different database configs and connection fn calls) can be easily wrapped and made generic. Same for readers, but anyway it's a different topic :)   

Option 2. Have the same cons as option 1. handling configs. 
Option 3. Same, handling configs. 

We fall slowly into env config hell and I would really consider config files (yaml, toml whatever). 
Regarding notifiers, if we don't want to refactor writers or move config to files, I think option 1. is what we are already doing and probably a way to go.  
If we are willing to refactor writers and merge main, we should then think about doing the same for notifiers (with config files).

Reply:
I agree. Regardless of this particular problem, we need to tackle configuration issues for just about all the services. We repeat NATS, Auth, HTTP(S) serve, and DB connection in almost every `main.go` and that's now quite a lot of code to maintain. It starts to hurt our codebase and I think that we should start resolving this issue ASAP.

Reply:
I will open a separate issue, for discussion that addressing this config problem.